aten/src/ATen/cpu/vec256/vec256_bfloat16.h

```diff
@@ -133,7 +133,8 @@ template <> class Vec256<BFloat16> {
       const Vec256<BFloat16>& b, const Vec256<BFloat16>& mask) {
     return _mm256_blendv_epi8(a.values, b.values, mask.values);
   }
-  static Vec256<BFloat16> arange(BFloat16 base = 0.f, BFloat16 step = 1.f) {
+  template<typename step_t>
+  static Vec256<BFloat16> arange(BFloat16 base = 0.f, step_t step = static_cast<step_t>(1)) {
```

Comment:
Why don't you do `step_t step = 1`? For all but most exotic cases if you can static_cast it to step_t, you can implicitly convert it to step_t.

Reply:
If I do `step_t step = 1`, then `step` would not be implicitly converted to type `step_t` -- `step_t` would default to `int`.

Reply:
You are either explicitly instantiating template (in which case step_t would be whatever you specified it to be), or implicitly instantiating it by passing some argument, in which case the default value of the argument does not matter. You cannot implicitly instantiate it with default value here, it will be a compile-time error (cannot infer step_t or some such). But since this construct is all over the place for all other types, ok, let it be here.